src/widget/wfindonmenu.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <QAction>
+#include <QMenu>
+#include <QUrlQuery>
+
+class Track;
+
+class WFindOnMenu : public QMenu {
+    Q_OBJECT
+  public:
+    explicit WFindOnMenu(
+            QWidget* parent = nullptr);
+    ~WFindOnMenu() override = default;
+
+    enum class Services {
+        SoundCloud,
+        LastFm,
+        Discogs
+    };
+
+    void openTheBrowser(const QString& serviceUrl,
```

Comment:
openInBrowser?

The serviceUrl should not be part of the public API.

Reply:
I think this function can be moved to the anonymous namespace in the cpp file, because it does not use any members from "this"

Reply:
I tried to make the URL private. But since I am using this method in WTrackMenu, this is a bit difficult. Or is it too easy and am I missing something?

I made private the Url in an another way. I defined a function in anonymous namespace that can return URL's according to chosen service, I am calling this function in openInBrowser, to return the urls from anonymous namespace. This time openInBrowser gets 2 arguments, which is the service and the query. But this time I needed to make "enum Service" public. So I am not sure which should be public and the other one private, is it okay to make Enum Service public?